mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/service/LuceneService.java

```diff
@@ -0,0 +1,125 @@
+package de.brandwatch.minianalytics.mentiongenerator.service;
+
+import de.brandwatch.minianalytics.mentiongenerator.kafka.Producer;
+import de.brandwatch.minianalytics.mentiongenerator.model.Mention;
+import de.brandwatch.minianalytics.mentiongenerator.model.Resource;
+import de.brandwatch.minianalytics.mentiongenerator.postgres.cache.QueryCache;
+import de.brandwatch.minianalytics.mentiongenerator.postgres.model.Query;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.TextField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.RAMDirectory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+@Service
+public class LuceneService {
+
+    private static final Logger logger = LoggerFactory.getLogger(LuceneService.class);
+
+    private final Producer producer;
+
+    private final QueryCache queryCache;
+
+    @Autowired
+    public LuceneService(Producer producer, QueryCache queryCache) throws IOException {
```

Comment:
This is not a _rule_, but there are some ordering typically used, for example first having static final, then final, then non final fields, and then the constructor, and then the methods. If everyone follows that, code can be read much faster. Here it's all a little mixed up, maybe you can change that.

Reply:
I always try to order everything 'logically' (for me at least), but I have changed it.

Reply:
You don't have to change everything that's suggested. You could as well argue why you think your ordering was better than what I suggested. There's no need to always just do it.

That being said though, the constructor was likely still misplaced.

Reply:
I think it's important, that a codebase is as uniform as possible, so that everybody can understand things quickly. So I change things to match the codebase even though I might think my ordering was better.